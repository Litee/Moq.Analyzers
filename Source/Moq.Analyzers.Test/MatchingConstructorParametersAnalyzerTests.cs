using ApprovalTests;
using ApprovalTests.Reporters;
using Microsoft.CodeAnalysis.Diagnostics;
using System.IO;
using TestHelper;
using Xunit;

namespace Moq.Analyzers.Test
{
    [UseReporter(typeof(DiffReporter))]
    public class MatchingConstructorParametersAnalyzerTests : DiagnosticVerifier
    {

        [Fact]
        public void ShouldFailIfClassParametersDoNotMatch()
        {
            Approvals.Verify(VerifyCSharpDiagnostic(File.ReadAllText("Data/MockClassWithParameters.cs")));
        }

        [Fact]
        public void ShouldPassIfCustomMockClassIsUsed()
        {
           // Approvals.Verify(VerifyCSharpDiagnostic(File.ReadAllText("Data/MockInterfaceWithParametersCustomMockFile.cs")));
        }

        protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
        {
            return new MatchingConstructorParametersAnalyzer();
        }
    }
}