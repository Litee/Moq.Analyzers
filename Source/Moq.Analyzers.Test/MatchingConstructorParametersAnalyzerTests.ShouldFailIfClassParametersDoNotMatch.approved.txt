Diagnostic 1
	Id: MOQ1003
	Location: SourceFile(Test0.cs[459..468))
	Code: (1, true)
	Severity: Warning
	Message: Parameters provided into mock do not match existing constructors.

Diagnostic 2
	Id: MOQ1003
	Location: SourceFile(Test0.cs[538..547))
	Code: (2, true)
	Severity: Warning
	Message: Parameters provided into mock do not match existing constructors.

Diagnostic 3
	Id: MOQ1003
	Location: SourceFile(Test0.cs[617..625))
	Code: ("1", 3)
	Severity: Warning
	Message: Parameters provided into mock do not match existing constructors.

Diagnostic 4
	Id: MOQ1003
	Location: SourceFile(Test0.cs[695..718))
	Code: (new int[] { 1, 2, 3 })
	Severity: Warning
	Message: Parameters provided into mock do not match existing constructors.

Diagnostic 5
	Id: MOQ1003
	Location: SourceFile(Test0.cs[813..843))
	Code: (MockBehavior.Strict, 4, true)
	Severity: Warning
	Message: Parameters provided into mock do not match existing constructors.

Diagnostic 6
	Id: MOQ1003
	Location: SourceFile(Test0.cs[917..946))
	Code: (MockBehavior.Loose, 5, true)
	Severity: Warning
	Message: Parameters provided into mock do not match existing constructors.

Diagnostic 7
	Id: MOQ1003
	Location: SourceFile(Test0.cs[1020..1048))
	Code: (MockBehavior.Loose, "2", 6)
	Severity: Warning
	Message: Parameters provided into mock do not match existing constructors.

